pkg/store/gcworker/gc_worker.go

```diff
@@ -823,55 +824,76 @@ func (w *GCWorker) deleteRanges(ctx context.Context, safePoint uint64, concurren
 		zap.String("uuid", w.uuid),
 		zap.Int("ranges", len(ranges)))
 	startTime := time.Now()
+
+	if concurrency < 1 {
+		concurrency = 1
+	}
+	concurrencyLimiter := make(chan struct{}, concurrency)
+	var cacheMu sync.Mutex
```

Comment:
How about wrap the `cacaheMu` together with the to be protected `gcPlacementRuleCache` as a placement rule GC context.

Or refactor the `doGCPlacementRules` implementation, make it a standalone util decoupling with `GCworker`, since the `(w *GCworker)` is not used in it at all.

Reply:
Maybe it's actually enough to use a `sync.Map` here. Though it's annoying that it's not generic.